File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        ListNode temp=head;
        int count=0,sum=0;
        while(temp!=null)
        {
            ++count;
            temp=temp.next;
        }
        while(head!=null)
        {
            sum=sum+head.val*(int)Math.pow(2,--count);
            head=head.next;
        }
        return sum;
    }
}

File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
       int count=0,sum=0;
        ListNode* temp=head;
        while(temp!=NULL)
            {++count;
            temp=temp->next;}
        while(head!=NULL)
        {
            sum=sum+head->val*pow(2,--count);
            head=head->next;
        }
        return sum;
    }
};